src/stream/src/executor/sink.rs

```diff
@@ -117,42 +120,69 @@ impl<F: LogStoreFactory> SinkExecutor<F> {
                 .any(|i| !self.sink_param.downstream_pk.contains(i))
         };
 
-        let write_log_stream = Self::execute_write_log(
-            self.input,
-            stream_key,
-            self.log_writer
-                .monitored(self.sink_writer_param.sink_metrics.clone()),
-            self.sink_param.sink_id,
+        let input = self.input.execute();
+
+        let input_row_count = self
+            .actor_context
+            .streaming_metrics
+            .sink_input_row_count
+            .with_guarded_label_values(&[
+                &sink_id.to_string(),
+                &actor_id.to_string(),
+                &self.actor_context.fragment_id.to_string(),
+            ]);
+
+        let input = input.inspect_ok(move |msg| {
+            if let Message::Chunk(c) = msg {
+                input_row_count.inc_by(c.capacity() as u64);
+            }
+        });
+
+        let processed_input = Self::process_msg(
+            input,
             self.sink_param.sink_type,
-            self.actor_context.clone(),
+            stream_key,
             stream_key_sink_pk_mismatch,
         );
 
-        dispatch_sink!(self.sink, sink, {
-            let consume_log_stream = Self::execute_consume_log(
-                sink,
-                self.log_reader,
-                self.input_columns,
-                self.sink_writer_param,
-                self.actor_context,
-                self.info,
-            );
-            select(consume_log_stream.into_stream(), write_log_stream).boxed()
-        })
+        if self.sink.is_sink_into_table() {
+            processed_input.boxed()
```

Comment:
Does it mean that log store will be absent for `SINK INTO TABLE`? Why?

Reply:
IIUC, the messages for `sink into table` is passed to the downstream table through the yielded executor output rather than writing to log store. The log store is only for decoupling with the external sink. The current variant `SinkImpl::TableSink` is a trivial sink with the same implementation as blackhole sink and does nothing.

I have updated the PR description for motivation about this PR.

Reply:
I see. Given that `TableSink` is just a trivial sink right now, it makes sense to not enable log store. Thanks for the explanation in the PR description!


Not related to this PR: failure isolation is also important for `SINK INTO TABLE` so I think at some point in the future we still need to add log store back for `TableSink`.

Reply:
> failure isolation is also important for SINK INTO TABLE so I think at some point in the future we still need to add log store back for TableSink

I guess we should solve it with https://github.com/risingwavelabs/rfcs/pull/84 instead of the log store in the sink executor. The decoupling log store should be in the source side of the sink's stream job instead of the sink's target side.